provider/diag/diagnostics.go

```diff
@@ -152,6 +152,24 @@ func (diags Diagnostics) CountBySeverity(sev Severity, includeSquashed bool) uin
 	return count
 }
 
+// BySeverity returns a subset of diagnostics matching the given severity.
+func (diags Diagnostics) BySeverity(sevs ...Severity) Diagnostics {
```

Comment:
Nice 🚀 

I actually started something like:
```go
func (diags Diagnostics) Filter(filterFunc func(d Diagnostic) bool) Diagnostics {
	res := make(Diagnostics, 0, len(diags))
	for i := range diags {
		if filterFunc(diags[i]) {
			res = append(res, diags[i])
		}
	}
	return res
}
```

But that could be too generic

Reply:
Thought about that, maybe generics can solve it later :)